tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -0,0 +1,114 @@
+
+import jakarta.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.MailetConfig;
+import org.apache.mailet.MailetException;
+
+import com.google.common.base.Strings;
+
+public class AIBotConfig {
+    public static String API_KEY_PARAMETER_NAME = "apiKey";
+    public static String GPT_ADDRESS_PARAMETER_NAME = "gptAddress";
+    public static String MODEL_PARAMETER_NAME = "model";
+    public static String BASE_URL_PARAMETER_NAME = "baseURL";
+
+    public static final LlmModel DEFAULT_LLM_MODEL =
+            new LlmModel(Llm.OPEN_AI, "gpt-4o-mini");
+
+    private final String apiKey;
+    private final String baseURL;
+    private final MailAddress gptAddress;
+    private final LlmModel llmModel;
+
+    public AIBotConfig(String apiKey, String baseURL, MailAddress gptAddress, LlmModel llmModel) {
+        Objects.requireNonNull(apiKey);
+        Objects.requireNonNull(gptAddress);
+        Objects.requireNonNull(llmModel);
+
+        this.apiKey = apiKey;
+        this.baseURL = baseURL;
+        this.gptAddress = gptAddress;
+        this.llmModel = llmModel;
+    }
+
+    public static AIBotConfig fromMailetConfig(MailetConfig mailetConfig) throws MailetException {
+        String apiKeyParam = mailetConfig.getInitParameter(API_KEY_PARAMETER_NAME);
+        String gptAddressParam = mailetConfig.getInitParameter(GPT_ADDRESS_PARAMETER_NAME);
+        String llmModelParam = mailetConfig.getInitParameter(MODEL_PARAMETER_NAME);
+        String baseUrlParam = mailetConfig.getInitParameter(BASE_URL_PARAMETER_NAME);
+
+        if (Strings.isNullOrEmpty(apiKeyParam)) {
+            throw new MailetException("No value for " + API_KEY_PARAMETER_NAME + " parameter was provided.");
+        }
+
+        if (Strings.isNullOrEmpty(gptAddressParam)) {
+            throw new MailetException("No value for " + GPT_ADDRESS_PARAMETER_NAME + " parameter was provided.");
+        }
+        
+        MailAddress mailAddress;
+        try {
+            mailAddress = new MailAddress(gptAddressParam);
+        } catch (AddressException e) {
+            throw new RuntimeException(e);
+        }
+
+        LlmModel llmModel;
+        if (Strings.isNullOrEmpty(llmModelParam)) {
+            llmModel = DEFAULT_LLM_MODEL;
+        } else {
+            llmModel = parseLlmModelParamOrThrow(llmModelParam);
+        }
+
+        return new AIBotConfig(apiKeyParam, baseUrlParam, mailAddress, llmModel);
+    }
+
+    private static LlmModel parseLlmModelParamOrThrow(String llmModelParam) throws MailetException {
+        String[] tokens = llmModelParam.split(",");
+
+        if (tokens.length != 2) {
+            throw new MailetException("""
+                    Invalid LLM model parameter. Please add the LLM along with the specific model to use.
+                    Example: openai,gpt-4-mini
+                    Provided value: %s""".formatted(llmModelParam));
+        }
+
+        String llm = tokens[0];
```

Comment:
Not used as far as I can tell except in the exception. Please drop the llm part of the model.

Reply:
It's used in `ChatLanguageModelFactory`, we need it to decide what implementation of ChatLanguageModel to use. For now, we only use OpenAIChatLangaugeModel, in the future more implementation would be used.

Reply:
Done.